entity_gym/entity_gym/environment.py

```diff
@@ -184,7 +203,67 @@ class ObsBatch:
     end_of_episode_info: Dict[int, EpisodeStats]
 
 
-def batch_obs(obs: List[Observation]) -> ObsBatch:
+def merge_obs(a: Optional[ObsBatch], b: ObsBatch) -> ObsBatch:
```

Comment:
I'd make both of the input args non-optional and check for none outside of the function.